tests/integration/cluster_fixture.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import copy
+import os
+import threading
+from uuid import uuid4, uuid1
+
+from trivup.clusters.KafkaCluster import KafkaCluster
+
+from confluent_kafka import Producer, Consumer
+
+from confluent_kafka.admin import AdminClient, NewTopic
+
+
+class ClusterFixture(object):
+    _slots_ = ()
```

Comment:
single underscore here, double underscore in the next class.
set here, list in the next class.

Why the difference?

Also: I'm still skeptical of specifying slots, from what I understand it is mainly useful for memory allocation reasons where you have a large number of small objects. Apart from that it just limits use/extensibility.

Reply:
It's primary purpose is actually to speed up attribute lookups. It's secondary use is memory savings. I'm not sure I understand how it limits extensibility if you want dynamic attributes, you can subclass this and not use slots.

Reply:
Neither seems relevant here though, I don't see how it adds any value to a class like this.
But it is up to you to keep it.

Reply:
Mostly I was experimenting with ways to maintain backward compatibility with different forms of encapsulation. I'll remove this entirely and use kafka_cluster within conftest.py as a factory method.